magento/graphql-ce
# Correct max page size for elasticsearch5 engine

### Description (*)
This value can be changed but can never exceed the value set for `index.max_result_window` in
elasticsearch. This looks like it was caught and fixed for the `elasticsearch` https://github.com/magento/magento2/commit/29d3b357723c763a1e91d567b135508a77fd0c96 engine, but
not for `elasticsearch5`

### Fixed Issues (if relevant)
1. magento/graphql-ce#305 : Product Search Error with Elasticsearch

### Manual testing scenarios (*)
1. Install Magento and configure search index using `elasticsearch5`
2. Execute graphql query such as:
```graphql
{
    products (
        search: "configurable"
        pageSize: 20
        currentPage: 1
        sort: {
            name: DESC
        }
    )
    {
        items {
            sku
            name
        }
    }
}
```
3. The above query should result in a valid response and not an error similar to the one reported in magento/graphql-ce#305

### Contribution checklist (*)
